client/src/pages/FilmDetails.tsx

```diff
@@ -24,7 +28,32 @@ export function FilmDetailPage() {
       }
     }
     getDetails();
-  }, [filmId]);
+
+    async function getWishlist() {
+      if (user) {
+        try {
+          const wishlistReq = {
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: `Bearer ${readToken()}`,
+            },
+          };
+          const wishlistResp = await fetch(
```

Comment:
Don't you have a file where all your fetch calls are? If not, you should create one. This useEffect is now getting pretty long and would benefit from moving the fetch calls into separate functions.